### PR TITLE
Remove delays from Acquire Channels

### PR DESCRIPTION
**Pull request by zachschoenfeld33**
### Summary

This PR removes pulse `delay` instructions from the qobj if they are on `Acquire` channels. They are unnecessary as pulse instructions are already absolutely timed. 

Additionally, an error is included when `Acquire` channels are used on `Play`.

The net result is that pulses excluding the `Acquire` instruction will not be passed into the qobj.

### CODE REVIEW

File: qiskit/assembler/assemble_schedules.py [hunk 1]
```diff
@@ -203,6 +203,11 @@ def _assemble_instructions(
                 name=name)
             user_pulselib[name] = instruction.pulse.samples
 
+        # ignore delay instrs on acq channels (don't matter w/ absolute timing)
```
Comment (lcapelluto) [inline qiskit/assembler/assemble_schedules.py:206]: This is a little misleading, since delays at the end of a sub-program do have an effect. This might be one case where a more detailed explanation (describing what would happen if this stayed) in the comments is warranted

File: qiskit/assembler/assemble_schedules.py [hunk 2]
```diff
@@ -203,6 +203,11 @@ def _assemble_instructions(
...
+        if (isinstance(instruction, instructions.Delay) and
+                isinstance(instruction.channel, channels.AcquireChannel)):
```
Comment (lcapelluto) [inline qiskit/assembler/assemble_schedules.py:208]: If the comment was true, and delays didn't matter with absolute timing, then this half of the condition could be removed, and we could ignore all Delays

File: qiskit/pulse/instructions/play.py [hunk 3]
```diff
@@ -47,6 +47,9 @@ def __init__(self, pulse: Pulse,
         """
         if not isinstance(pulse, Pulse):
             raise PulseError("The `pulse` argument to `Play` must be of type `library.Pulse`.")
+        if not isinstance(channel, PulseChannel):
```
Comment (lcapelluto) [inline qiskit/pulse/instructions/play.py:50]: Where did this come up?

File: test/python/pulse/test_instructions.py [hunk 4]
```diff
@@ -191,19 +191,28 @@ def test_default(self):
 class TestPlay(QiskitTestCase):
     """Play tests."""
 
+    def setUp(self):
```
Comment (lcapelluto) [inline test/python/pulse/test_instructions.py:194]: How are these changes related to the bug?

**Review by lcapelluto — CHANGES_REQUESTED** [review]

File: qiskit/pulse/instructions/play.py (hunk 3 above)
Comment (zachschoenfeld33) [reply qiskit/pulse/instructions/play.py:50]: It's not directly related, but I saw it and so wanted to make sure it was addressed.

File: test/python/pulse/test_instructions.py (hunk 4 above)
Comment (zachschoenfeld33) [reply test/python/pulse/test_instructions.py:194]: Not directly related, but the current code will allow you to play pulses on `AcquireChannel`'s, which should be illegal.

File: qiskit/assembler/assemble_schedules.py (hunk 1 above)
Comment (zachschoenfeld33) [reply qiskit/assembler/assemble_schedules.py]: Addressed in https://github.com/Qiskit/qiskit-terra/pull/6000/commits/e24adf163f5804576f2d85c7c5bf8a895238775b.

File: qiskit/assembler/assemble_schedules.py (hunk 2 above)
Comment (zachschoenfeld33) [reply qiskit/assembler/assemble_schedules.py:209]: Addressed in https://github.com/Qiskit/qiskit-terra/pull/6000/commits/e24adf163f5804576f2d85c7c5bf8a895238775b.

**Review by lcapelluto — APPROVED** [review]
